tests/proc/test_run.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+from subprocess import CalledProcessError
+from unittest.mock import patch
+
+import pytest
+
+from devenv.lib.proc import base_env
+from devenv.lib.proc import run
+
+
+def test_run_with_stdout() -> None:
+    cmd = ("echo", "Hello, World!")
+    expected_output = "Hello, World!"
+    user_environ = {"PATH": "/usr/local/bin:/usr/bin"}
+    env = {**user_environ, **base_env}
+
+    with patch("devenv.lib.proc.subprocess_run") as mock_run, patch(
+        "devenv.lib.proc.constants.user_environ", user_environ
+    ):
+        mock_run.return_value.stdout = expected_output.encode()
+        result = run(cmd, stdout=True)
+
+    assert result == expected_output
+    mock_run.assert_called_once_with(
+        cmd, check=True, stdout=-1, cwd=None, env=env
+    )
+
+
+def test_run_without_stdout() -> None:
+    cmd = ("echo", "Hello, World!")
+    user_environ = {"PATH": "/usr/local/bin:/usr/bin"}
+    env = {**user_environ, **base_env}
+
+    with patch("devenv.lib.proc.subprocess_run") as mock_run, patch(
+        "devenv.lib.proc.constants.user_environ", user_environ
+    ):
+        result = run(cmd)
+
+    assert result is None
+    mock_run.assert_called_once_with(
+        cmd, check=True, stdout=None, cwd=None, env=env
+    )
+
+
+def test_run_with_pathprepend() -> None:
+    cmd = ("echo", "Hello, World!")
+    pathprepend = "/path/to/bin"
+    user_environ = {"PATH": "/usr/local/bin:/usr/bin"}
+    env = {**user_environ, **base_env}
+    env["PATH"] = f"{pathprepend}:{env['PATH']}"
+
+    with patch("devenv.lib.proc.subprocess_run") as mock_run, patch(
+        "devenv.lib.proc.constants.user_environ", user_environ
+    ):
+        mock_run.return_value.stdout = None
+        result = run(cmd, pathprepend=pathprepend)
+
+    assert result is None
+    mock_run.assert_called_once_with(
+        cmd, check=True, stdout=None, cwd=None, env=env
+    )
+
+
+def test_run_command_not_found() -> None:
+    cmd = ("invalid_command",)
+    user_environ = {"PATH": "/usr/local/bin:/usr/bin"}
+    env = {**user_environ, **base_env}
+
+    with patch("devenv.lib.proc.subprocess_run") as mock_run, patch(
```

Comment:
can we actually run subprocess_run wherever possible? i didn't mean to only change that one test, i meant it generally